src/ducks/sharing/WhoHasAccess.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { connect } from 'react-redux'
+
+import Recipient from '../../components/Recipient'
+import { getRecipients } from '.'
+
+const WhoHasAccess = ({ recipients }) => (
+  <div>
+    {recipients.map(({email, url, status}) => (
+      <Recipient name={email} url={url} status={status} />
+    ))}
+  </div>
+)
+
+class StatefulWhoHasAccess extends React.Component {
+  state = {
+    recipients: []
+  }
+
+  componentDidMount () {
+    const { _id, _type } = this.props.document
+    const { t } = this.context
+    getRecipients(_id, _type)
+    .then(recipients =>
+      recipients.map(recipient => {
+        const status = recipient.status === 'accepted'
+         ? t(`Share.status.${recipient.status}.${recipient.type}`)
+         : t(`Share.status.${recipient.status}`)
+        return {
+          ...recipient,
+          status
+        }
+      })
+    )
+    .then(recipients => {
+      this.setState(state => ({ ...state, recipients }))
+    })
+    .catch(console.error.bind(console))
```

Comment:
I don't know what to do with errors like this one. Do we want to handle and shut it down? Do we want to handle and display it to user?
What do you think?

This line is to avoid a big fat error in the console in case of _Promise rejection unhandled_.